app/controllers/application_controller.rb

```diff
@@ -2081,6 +2081,10 @@ def identify_tl_or_perf_record
   end
 
   def assert_privileges(feature)
+    if MiqProductFeature.my_root_tenant_identifier?(feature) && params.key?(:id)
```

Comment:
`assert_privileged` was a rather clean method, not a pure function as it talks to a database.

Not wou are adding a side channel to it -- you talk to `params`. This goes directly against the cleanup efforts that we are having.